### PR TITLE
Add support for selecting inspector node from a query parameter

### PR DESCRIPTION
**Pull request by kenzieschmoll**
A specific widget in the inspector can be selected by passing the valueId as a query parameter:
e.g. `&inspectorRef=inspector-7`

### CODE REVIEW

File: packages/devtools_app/lib/src/inspector/inspector_controller.dart [hunk 1]
```diff
@@ -581,42 +580,56 @@ class InspectorController extends DisposableController
...
-        ? group.getSelection(selectedDiagnostic, treeType, isSummaryTree: false)
-        : null;
+    if (firstFrame) {
+      final queryParams = loadQueryParams();
+      if (queryParams.containsKey(inspectorRefQueryParam)) {
+        print('trying to select from service');
+        await group.setSelectionInspector(
+          InspectorInstanceRef(queryParams[inspectorRefQueryParam]),
+          false,
+        );
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/inspector/inspector_controller.dart:591]: @jacob314 this code as is results in some errors from the flutter framework:
```
{"exception":"Id does not exist.","stack":"#0      WidgetInspectorService.toObject (package:flutter/src/widgets/widget_inspector.dart:1286:7)\n#1      WidgetInspectorService.setSelectionById (package:flutter/src/widgets/widget_inspector.dart:1348:25)\n#2      WidgetInspectorService._registerServiceExtensionWithArg.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:867:35)\n#3      WidgetInspectorService._registerServiceExtensionWithArg.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:864:17)\n#4      BindingBase.registerServiceExtension.<anonymous closure> (package:flutter/src/foundation/binding.dart:580:32)\n<asynchronous suspension>\n","method":"ext.flutter.inspector.setSelectionById"}
trying to select from service
Error: ext.flutter.inspector.setSelectionById: (-32000) Server error
{"exception":"Id does not exist.","stack":"#0      WidgetInspectorService.toObject (package:flutter/src/widgets/widget_inspector.dart:1286:7)\n#1      WidgetInspectorService.setSelectionById (package:flutter/src/widgets/widget_inspector.dart:1348:25)\n#2      WidgetInspectorService._registerServiceExtensionWithArg.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:867:35)\n#3      WidgetInspectorService._registerServiceExtensionWithArg.<anonymous closure> (package:flutter/src/widgets/widget_inspector.dart:864:17)\n#4      BindingBase.registerServiceExtension.<anonymous closure> (package:flutter/src/foundation/binding.dart:580:32)\n<asynchronous suspension>\n","method":"ext.flutter.inspector.setSelectionById"}
```

Comment (jacob314) [reply packages/devtools_app/lib/src/inspector/inspector_controller.dart:591]: Can you print the id you are passing?

Comment (jacob314) [reply packages/devtools_app/lib/src/inspector/inspector_controller.dart:591]: You can also add logging to the WidgetInspectorService code to make sure that the id that got read in is what you expected.

File: packages/devtools_app/lib/src/inspector/inspector_controller.dart [hunk 2]
```diff
@@ -581,42 +580,56 @@ class InspectorController extends DisposableController
...
+      }
+    } else {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/inspector_controller.dart:593]: this shouldn't be an else. You should read the selection from the flutter device after you set the selection on the device. That will make everything work without you needing to duplicate logic for separating out the details and summary selections, etc.

File: packages/devtools_app/lib/src/inspector/inspector_controller.dart [hunk 3]
```diff
@@ -581,42 +580,56 @@ class InspectorController extends DisposableController
...
+        isSummaryTree: isSummaryTree,
+      );
 
-    try {
-      final RemoteDiagnosticsNode newSelection = await pendingSelectionFuture;
-      if (group.disposed) return;
-      RemoteDiagnosticsNode detailsSelection;
+      final Future<RemoteDiagnosticsNode> pendingDetailsFuture = isSummaryTree
+          ? group.getSelection(selectedDiagnostic, treeType,
+              isSummaryTree: false)
```
Comment (jacob314) [inline packages/devtools_app/lib/src/inspector/inspector_controller.dart:602]: nit: add trailing comma.

File: packages/devtools_app/lib/src/inspector/inspector_controller.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/inspector/inspector_controller.dart:591]: I am passing an id like: `instance-inspector-7`

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/inspector/inspector_controller.dart:590]: update - issue was with the "instance-" prefix in the id

File: packages/devtools_app/lib/src/inspector/inspector_controller.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/inspector/inspector_controller.dart]: done

**Review by jacob314 — APPROVED** [review]
lgtm

**Comment by kenzieschmoll** [comment]
part of https://github.com/flutter/devtools/issues/2504

**Pull request merged**